### PR TITLE
chore: update windows server ci version from 2019 to 2022

### PR DESCRIPTION
**Pull request by KolbyML**
### What was wrong?
CI broke for windows compiling c-kzg-4844 in pr https://github.com/ethereum/trin/pull/942
### How was it fixed?
By updating our windows server version from 2019 to 2022

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -171,20 +171,13 @@ jobs:
...
-      - run:
-          name: Add target
-          command: rustup target add x86_64-pc-windows-msvc
-      - run:
-          name: Install target toolchain
-          command: rustup toolchain install stable-x86_64-pc-windows-msvc
       - run:
           name: Check Trin workspace
-          command: cargo check --workspace
+          # We are running all these together because this version of circleci windows has an environment variable bug
```
Comment (carver) [inline .circleci/config.yml:176]: Can you link to an explanation of the bug, or describe a bit more what's happening here if no reference is available? When we find this comment later, we might want to evaluate if the bug has been resolved and we can split the steps back out. We will appreciate if we leave ourselves more hints/direction on how to do that.

**Review by carver — APPROVED** [review]
Thanks for hunting this down!

**Comment by carver** [comment]
Hm, I tried this patch on #942 and it still failed. Haven't looked at why, yet

File: .circleci/config.yml (hunk 1 above)
Comment (KolbyML) [reply .circleci/config.yml:176]: resolved

**Pull request merged**